File: benchmarks/fixtures/billing/src/Acme.Billing/Tax.cs
namespace Acme.Billing;

public static class Tax
{
	private static readonly Dictionary<string, decimal> Rates = new(StringComparer.OrdinalIgnoreCase)
	{
		["PT"] = 0.23m,
		["ES"] = 0.21m,
		["DE"] = 0.19m,
		["FR"] = 0.20m,
	};

	public static decimal Rate(string? region) =>
		Rates.TryGetValue(region ?? "", out var r) ? r : 0m;

	public static decimal CalcTax(decimal amount, string? region) =>
		Money.Round2(amount * Rate(region));
}

File: benchmarks/fixtures/shop/src/pricing/tax.js
'use strict';

const { round2 } = require('../util/money');

const RATES = { PT: 0.23, ES: 0.21, DE: 0.19, FR: 0.2 };

function taxRate(region) {
	return RATES[String(region || '').toUpperCase()] ?? 0;
}

function calcTax(amount, region) {
	return round2(amount * taxRate(region));
}

module.exports = { RATES, taxRate, calcTax };
